src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/RefundItemsTransaction.php

```diff
@@ -0,0 +1,168 @@
+     */
+    protected $moneyFacade;
+
+    /**
+     * @var \SprykerEco\Zed\Braintree\Business\Payment\Transaction\Handler\ShipmentRefundTransactionHandlerInterface
+     */
+    protected $shipmentRefundTransactionHandler;
+
+    /**
+     * @var \SprykerEco\Zed\Braintree\Persistence\BraintreeRepositoryInterface
+     */
+    protected $braintreeRepository;
+
+    /**
+     * @param \SprykerEco\Zed\Braintree\BraintreeConfig $brainTreeConfig
+     * @param \SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToMoneyFacadeInterface $moneyFacade
+     * @param \SprykerEco\Zed\Braintree\Business\Payment\Transaction\Handler\ShipmentRefundTransactionHandlerInterface $shipmentRefundTransactionHandler
+     * @param \SprykerEco\Zed\Braintree\Persistence\BraintreeRepositoryInterface $braintreeRepository
+     */
+    public function __construct(
+        BraintreeConfig $brainTreeConfig,
+        BraintreeToMoneyFacadeInterface $moneyFacade,
+        ShipmentRefundTransactionHandlerInterface $shipmentRefundTransactionHandler,
+        BraintreeRepositoryInterface $braintreeRepository
+    ) {
+        parent::__construct($brainTreeConfig);
+        $this->moneyFacade = $moneyFacade;
+        $this->shipmentRefundTransactionHandler = $shipmentRefundTransactionHandler;
+        $this->braintreeRepository = $braintreeRepository;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getTransactionType(): string
+    {
+        return ApiConstants::CREDIT;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getTransactionCode(): string
+    {
+        return ApiConstants::TRANSACTION_CODE_REFUND;
+    }
+
+    /**
+     * @return \Braintree\Result\Error|\Braintree\Result\Successful
+     */
+    public function doTransaction()
+    {
+        return $this->refund();
+    }
+
+    /**
+     * @param \Braintree\Result\Error|\Braintree\Result\Successful $response
+     *
+     * @return \Generated\Shared\Transfer\BraintreeTransactionResponseTransfer
+     */
+    public function afterTransaction($response): BraintreeTransactionResponseTransfer
+    {
+        $shipmentExpenseTransfer = $this->getShipmentExpenseTransfer();
+
+        if ($shipmentExpenseTransfer) {
+            $this->refundShipmentExpense();
+        }
+
+        return parent::afterTransaction($response);
+    }
+
+    /**
+     * @return \Braintree\Result\Error|\Braintree\Result\Successful
+     */
+    protected function refund()
+    {
+        $transaction = $this->findTransaction();
+
+        if ($transaction->status === ApiConstants::STATUS_CODE_CAPTURE_SUBMITTED) {
+            return BraintreeTransaction::void($this->getTransactionIdentifier());
+        }
+
+        return BraintreeTransaction::refund(
+            $this->getTransactionIdentifier(),
+            $this->getAmount()
+        );
+    }
+
+    /**
+     * @return float|null
+     */
+    protected function getAmount(): ?float
```

Comment:
`MoneyFacade::convertIntegerToDecimal()` returns only float. Why it can be null?